via/services/youtube.py

```diff
@@ -75,19 +87,32 @@ def get_video_title(self, video_id):
         except Exception as exc:
             raise YouTubeDataAPIError("getting the video title failed") from exc
 
-    def get_transcript(self, video_id):
-        """
-        Call the YouTube API and return the transcript for the given video_id.
+    def get_video_info(self, video_id=None, video_url=None) -> Video:
+        if video_url:
+            video_id = self.get_video_id(video_url)
```

Comment:
```suggestion
    def get_video_info(self, video_id) -> Video:
```

The `video_url` argument isn't needed: this is only called by `get_transcript()` below and it always passes `video_id` never `video_url`